minion/server/routes.py

```diff
@@ -10,25 +10,29 @@
 import re
 import mongoengine
 from flask import Blueprint, request, jsonify, render_template
+from jinja2.exceptions import UndefinedError
 
 from .models import Job, mongo_to_dict
 from .core import workers, stop_server, ensure_mongo
 import pymongo
 
 api = Blueprint("api", __name__)
 
+@ensure_mongo
 def get_jobs_page(page=1, page_size=10):
     """Returns all jobs from the given page."""
     return Job.objects.paginate(page=page, per_page=page_size)
 
 
 @api.route("/", methods=["GET"])
-@ensure_mongo
 def index():
     """Serve index HTML."""
     page = int(request.args.get("page", 1))
-    return render_template("index.html", jobs=get_jobs_page(page=page))
-
+    
+    try:
+        return render_template("index.html", jobs=get_jobs_page(page=page))
+    except UndefinedError:
```

Comment:
This is the exception raised by jinja/werkzeug. It is a bit generic, but since you will only get this error on the index view when something goes wrong with parsing the get_jobs_page it doesn't really matter.